C++/Navio/Common/gpio.cpp

```diff
@@ -9,12 +9,17 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
+#include <fstream>
+#include <string>
+#include <iostream>
 
 #include "gpio.h"
 
 #define LOW                 0
 #define HIGH                1
 
+#define RPI_MODEL_ERROR     1
```

Comment:
Well, the name is good now, thanks! But the `1` is bad. You actually have such a variant of RPi model. And your switch has such `case`. Think about how to fix it, please. As an easy decision - you can read [this](https://en.wikipedia.org/wiki/Hexspeak). We sometimes use it. But it will be good if you find a better solution.

Reply:
What if I would use `exit(EXIT_FAILURE);` instead of this define? I think it is much better to just exit the code if there's error.

Reply:
You can try, but:
- You should exit at only once place
- It's better to exit at a higher level

Reply:
What do you mean by "exit at a higher level"? 

I was testing it with a test file instead of model_path, using different possibilities and It seems to work fine. But if you say this is unsafe, I will try to find another method.

Reply:
Like if you have 3 functions a(), b(), c(), you call `b` and `c` from `a`, it's better to exit at `a` if smth goes wrong at `b` or `c`. If you will add exit directly to `b` and `c`, it will be harder to track where your program can exit. Harder to read and debug. And imagine one day you'll have to add 10 more functions with exits.
If that point doesn't seem critical to you, I have one more and it's a technical one - when you call exit & Co, you will not go through stack unwilling. It's a C function and it doesn't know anything about how CPP and RAII work. You'll get no destructors called. In our example, it means ifstream's state left is undefined (ofc Linux will take care of it, but nevertheless).
Safe options: `return`, `throw & catch`. The second one looks like overkill here and I don't think the `main` has an exception handling here, so I would suggest using returns. Just check them and find a way to tell other parts of the program that smth failed.

Reply:
Thank you for this explanation! This is my solution: I define `RPI_MODEL_ERROR` as 0 and just would check it in _init_ function. I guess this pretty easy way would be enough. Check it, please.